includes/Checker/Checks/Enqueued_Scripts_Size_Check.php

```diff
@@ -0,0 +1,229 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\Checks\Enqueued_Scripts_Size_Check
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker\Checks;
+
+use WordPress\Plugin_Check\Checker\Check_Result;
+use WordPress\Plugin_Check\Checker\With_Shared_Preparations;
+use WordPress\Plugin_Check\Traits\URL_Aware;
+use WordPress\Plugin_Check\Checker\Preparations\Demo_Posts_Creation_Preparation;
+use Exception;
+
+/**
+ * Check for running WordPress internationalization sniffs.
+ *
+ * @since n.e.x.t
+ */
+class Enqueued_Scripts_Size_Check extends Abstract_Runtime_Check implements With_Shared_Preparations {
+
+	use URL_Aware;
+
+	/**
+	 * Threshold for script size to surface a warning for.
+	 *
+	 * @since n.e.x.t
+	 * @var int
+	 */
+	private $threshold_size;
+
+	/**
+	 * List of viewable post types.
+	 *
+	 * @since n.e.x.t
+	 * @var array
+	 */
+	private $viewable_post_types;
+
+	/**
+	 * Set the threshold size for script sizes to surface warnings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param integer $threshold_size The threshold in bytes for script size to surface warnings.
+	 */
+	public function __construct( $threshold_size = 300000 ) {
+		$this->threshold_size = $threshold_size;
+
+		$this->viewable_post_types = array_filter(
+			get_post_types(),
+			function( $post_type ) {
+				return is_post_type_viewable( $post_type );
+			}
+		);
+	}
+
+	/**
+	 * Runs this preparation step for the environment and returns a cleanup function.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return callable Cleanup function to revert any changes made here.
+	 *
+	 * @throws Exception Thrown when preparation fails.
+	 */
+	public function prepare() {
+		$orig_scripts = isset( $GLOBALS['wp_scripts'] ) ? $GLOBALS['wp_scripts'] : null;
+
+		// Backup the original values for the global state.
+		$check = $this;
+		$check->backup_globals();
+
+		return function() use ( $orig_scripts, $check ) {
+			if ( is_null( $orig_scripts ) ) {
+				unset( $GLOBALS['wp_scripts'] );
+			} else {
+				$GLOBALS['wp_scripts'] = $orig_scripts;
+			}
+
+			$check->restore_globals();
```

Comment:
The assignment to a `$check` variable is unnecessary. Closures that are defined in a class can access `$this` to reference the class instance (as of PHP 5.4, which is lower than our minimum requirement).
```suggestion
		$this->backup_globals();

		return function() use ( $orig_scripts ) {
			if ( is_null( $orig_scripts ) ) {
				unset( $GLOBALS['wp_scripts'] );
			} else {
				$GLOBALS['wp_scripts'] = $orig_scripts;
			}

			$this->restore_globals();
```